libweb3jsonrpc/Test.cpp

```diff
@@ -23,16 +23,60 @@
 #include "Test.h"
 #include <jsonrpccpp/common/errors.h>
 #include <jsonrpccpp/common/exception.h>
-#include <libethereum/ClientTest.h>
+#include <libdevcore/CommonJS.h>
 #include <libethereum/ChainParams.h>
+#include <libethereum/ClientTest.h>
 
 using namespace std;
 using namespace dev;
+using namespace dev::eth;
 using namespace dev::rpc;
 using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+namespace
+{
+string logEntriesToLogHash(eth::LogEntries const& _logs)
+{
+    RLPStream s;
+    s.appendList(_logs.size());
+    for (eth::LogEntry const& l : _logs)
+        l.streamRLP(s);
+    return toJS(sha3(s.out()));
+}
+}
+
+string Test::test_getLogHash(string const& _txHash)
+{
+    try
+    {
+        h256 txHash;
+        try
+        {
+            txHash = h256(_txHash);
+        }
+        catch (BadHexCharacter const&)
+        {
+            throw JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS);
```

Comment:
There was no any sense really in using `BOOST_THROW_EXCEPTION` for `JsonRpcException` which is not a boost exception